Src/BootCamp.Chapter/TextTable.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    /// <summary>
+    /// Part 1.
+    /// </summary>
+    public class TextTable
+    {
+        /*
+
+         Input: "Hello", 0
+           +-----+
+           |Hello|
+           +-----+
+           
+         Input: $"Hello{Environment.NewLine}World!", 0
+           +------+
+           |Hello |
+           |World!|
+           +------+
+           
+         Input: "Hello", 1
+           +-------+
+           |       |
+           | Hello |
+           |       |
+           +-------+
+
+         */
+
+        /// <summary>
+        /// Build a table for given message with given padding.
+        /// Padding means how many spaces will a message be wrapped with.
+        /// Table itself is made of: "+-" symbolmessage. 
+        /// </summary>
+        public string Build(string message, int padding)
+        {
+            //Analyze message
+            if (String.IsNullOrEmpty(message))
+            {
+                return "";
+            }
+            var numOfMessageLines = GetNumberOfMessageLines(message);
+            var messageLine = GetMessageLines(message, numOfMessageLines);
+            var maxLength = GetMaxMessageLineLength(messageLine);
+
+            //Build table from message
+            return BuildTable(messageLine, maxLength, padding);
+        }
+
+        private int GetNumberOfMessageLines(string message)
+        {
+            var numOfLines = 1;
+
+            for (var i = 0; i < message.Length - 1; i++)
+            {
+                if (message[i] == '\r' && message[i + 1] == '\n')
+                {
+                    numOfLines++;
+                }
+            }
+
+            return numOfLines;
+        }
+
+        private string[] GetMessageLines(string message, int numOfMessageLines)
+        {
+            var messageLine = new string[numOfMessageLines];
+            var messageCount = 0;
+            var beginPos = 0;
+            var endPos = 0;
+
+            if (numOfMessageLines == 0 && message.Length > 0)
+            {
+                messageLine[0] = message;
+            }
+
+            for (var i = 0; i < message.Length - 1; i++)
+            {
+                if (message[i] == '\r' && message[i + 1] == '\n')
+                {
+                    endPos = i - 1;
+                    messageLine[messageCount++] = message.Substring(beginPos, endPos - beginPos + 1);
+                    beginPos = i + 2;
+                }
+            }
+
+            if (endPos <= message.Length)
+            {
+                messageLine[messageCount] = message.Substring(beginPos, message.Length - beginPos);
+            }
+
+            return messageLine;
+        }
+
+        private int GetMaxMessageLineLength(string[] messageLine)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make 'GetMaxMessageLineLength' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)